softlayer/resource_softlayer_ssh_key.go

```diff
@@ -28,11 +28,17 @@ func resourceSoftLayerSSHKey() *schema.Resource {
 				Computed: true,
 			},
 
-			"name": &schema.Schema{
+			"label": &schema.Schema{
 				Type:     schema.TypeString,
 				Required: true,
 			},
 
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Removed:  "Use 'label' instead",
+			},
+
```

Comment:
I would not bother with deprecations... Up to you. and it's nice that they have _Removed_ for these situations, but I don't think we are there yet to need it.